src/Raven.Server/Documents/Indexes/Persistence/Lucene/LuceneIndexPersistence.cs

```diff
@@ -106,6 +112,47 @@ public LuceneIndexPersistence(Index index)
                 string fieldName = field.Key;
                 _suggestionsIndexSearcherHolders[fieldName] = new IndexSearcherHolder(state => new IndexSearcher(_suggestionsDirectories[fieldName], true, state), _index._indexStorage.DocumentDatabase);
             }
+
+            IndexSearcher CreateIndexSearcher(IState state)
+            {
+                lock (this)
+                {
+                    var reader = _lastReader;
+
+                    if (reader != null)
+                    {
+                        if (reader.RefCount > 0)
+                        {
+                            try
+                            {
+                                var newReader = reader.Reopen(state);
+                                if (newReader != reader)
+                                    reader.DecRef(state);
+
+                                reader = _lastReader = newReader;
+                            }
+                            catch (Exception e)
+                            {
+                                if (_logger.IsInfoEnabled)
+                                    _logger.Info($"Could not reopen the index reader for index '{_index.Name}'.", e);
+
+                                // fallback strategy in case of a reader to be closed
+                                // before Reopen and DecRef are executed
+                                reader = null;
+                            }
+                        }
+                        else
+                            reader = null;
```

Comment:
Invert the `if` and add `{}` for balance?

Reply:
done